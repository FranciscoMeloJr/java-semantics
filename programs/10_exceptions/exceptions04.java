class main {
  main(String[] args) {
    int x = 1;
    try { int x = 5; throw x; int y = x/0;}   // division by zero unreachable
    catch(int y) {y = y + 2; System.out.print(y+" ");}   // new y here, the one declared by catch
    System.out.println(x);
    System.out.println("Done!");
  }
}

public class exceptions04 {
  public static void main(String[] args) {
    new main(args);
  }
}

// 7 1
// Done!
