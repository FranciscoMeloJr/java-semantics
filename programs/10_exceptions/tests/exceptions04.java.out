7 1
Done!
